### ISSUE TITLE
List of available / compatible modems

### ISSUE THREAD

**Issue by b-morgan** [thess/callattendant#26]
Is there a list of available / compatible modems? 

I have a Sentry Call Blocker that recently died and this project looks like an excellent replacement but finding a modem appears to be more difficult that I expected.

**Comment by thess** [comment]
Last time I looked, the US Robotics USR5637 mentioned in the doc has become a bit expensive. However, any modem with a Conexant CX930xx chipset should be 100% compatible. I do all my testing on a Conexant USB dongle from here:

https://www.amazon.com/gp/product/B01MYLE06I

**Comment by roycewilliams** [comment]
A couple of similar projects also keep more detailed lists of potentially compatible Conexant modems. I've linked to a couple of them here:

https://github.com/roycewilliams/callattendant-resources?tab=readme-ov-file#modem-hardware-ideas

It might be cool to build our own list, based on feedback of testing with callattendant itself by interested parties.

**Comment by whitedavidp** [comment]
I have a USR that I got smei-used off EBay for $40. I am happy with it but you MUST be sure its firmware is fully up-to-date.

I have no personal experience, but I did read [here](https://github.com/emxsys/callattendant/issues/139) about some issues with Zoom modems. So be careful.

**Comment by thess** [comment]
It would be interesting to know of anyone with a Zoom 3095 (Conexant) USB modem has tried this version of Callattendant. I made a number of changes / improvements to the handling the CNX930xx chips since that was what I have to work with.

@whitedavidp - Did you have to do a firmware update using the procedures on the USR site?

**Comment by whitedavidp** [comment]
> It would be interesting to know of anyone with a Zoom 3095 (Conexant) USB modem has tried this version of Callattendant. I made a number of changes / improvements to the handling the CNX930xx chips since that was what I have to work with.
> 
> @whitedavidp - Did you have to do a firmware update using the procedures on the USR site?
Yes, it would be interesting. Maybe someone having trouble on that thread still has one???

I am not sure if I HAD to do a firmware update. But the unit I got was not on the latest firmware. And I had read elsewhere that the update was basically a requirement. So yes, I got the file and procedure from the [USR site](https://www.usr.com/en-support/product/?prod=5637). On my windows 10 machine, it was a snap!

**Comment by RenOfHeavens** [comment]
> It would be interesting to know of anyone with a Zoom 3095 (Conexant) USB modem has tried this version of Callattendant. I made a number of changes / improvements to the handling the CNX930xx chips since that was what I have to work with.

I have the "square" Zoom 3095 (USB ID 0803:3095), and a generic conexant one (USB ID 0572:1329). They both work fine for me with a couple of initialisation commands: `AT+GCI=B4;AT+VCID=1;AT!4886=00`

My provider is strange in that they use Bell FSK (AT!4886=00), which isn't common here

**Comment by gwvr** [comment]
I've just got callattendant up and running with a Dynamode M56EXT-USB-T, which is identified as
> Bus 001 Device 002: ID 0572:1329 Conexant Systems (Rockwell), Inc. USB Modem

I'm in the UK. I've found that `OPTIONAL_MODEM_INIT = "AT-SCID=1"` is required for caller id. I don't think we can get caller name in the UK. NCID was useful for testing different modem init strings.

The modem was only £22, and is connected to a Pi Zero 2 W, so I consider this a cheap experiment.

**Comment by LESchwartz1959** [comment]
Here's another [inexpensive modem](https://www.amazon.com/dp/B008RZTJC0) from Amazon.  And yet [another one](https://www.amazon.com/External-Compatible-Auto-Dial-Dialing-Retrain/dp/B0BR5VKYR7/ref=sr_1_5_sspa?crid=V4XGMT6GDOB8).  I have personally tested both with CA on Raspberry Pi.

Larry

**Comment by usemi25** [comment]
Can you say what you did to get those to work?  I bought one and have been totally unable to get the driver to compile - just a cascade of failures, starting with hardcoded links to kernel sources that are out of date, but after fixing those, other issues that Gemini says are not fixable other than by going back to an old version (e.g., Ubuntu 12 or 14).  Gemini recommends using a hardware controller-based modem like the USR5637; I had not realized that the USR, Zoom, and a few others, had hardware controllers and would be plug-and-play, while the newer, cheap ones were software controller-based and, as far as I can tell, have ancient drivers that can't be built with a modern Linux distro.

As an addendum, working with Claude, I finally got the modem driver to compile under Gallium OS (kernel 4.16.18, closer to what the driver was built for, not 7.X with current Xubuntu), but it is not functioning - no modem seen.  So if someone knows how to get a softmodem with a CX93010 chip to run, please post.

Thanks!

**Comment by roycewilliams** [comment]
Yep, my understanding is that only hard modems are supported.

**Comment by LESchwartz1959** [comment]
> Can you say what you did to get those to work?

I didn't need to do anything special to get them working on Raspberry PI, I just installed CA and plugged the modems in and they worked.  I tested on both Pi 3 & 5 with the standard Pi OS.  There was no need for me to recompile the modem driver, etc.

Larry

**Comment by usemi25** [comment]
Ok, I got CA working, finally, with the modem I bought (off eBay, not Amazon, but looked the same as the Amazon one, even down to the photo of the software disc).  First I wasted tons of time trying to get a Zoom 3095 I bought from eBay in "open box" condition to work.  First, I learned that before CA could access the modem, I had permit access - I just did "chmod 777 /dev/ttyACM0" once I saw that was how it showed up. The default privileges assigned by Xubuntu were something like RW for root and nothing for everyone else. CA then sorta, kinda, intermittently worked. Sometimes let calls through instead of screening them, even though the log said a ring was detected. Just letting it sit for a long time (e.g., overnight) would then find it working. But VM absolutely would not work, though "record_message" would - but no "press 0" option that way.

I finally decided to try the eBay modem again, realizing I hadn't known that I had to do the chmod before CA could access it.  I did that, and everything, including VM, works like a charm.

I don't know if I just got a flaky Zoom 3095 or that's a general issue with them.  But I can now also confirm the relatively inexpensive, new Connexant 93010-powered USB modems off Amazon and eBay work fine.  I don't know if everyone has to do the chmod after plugging it in (or whether "sudo callattendant" would also get around that).  But it might be helpful to put a line or two into the setup instructions to save others all my headaches.  Also, putting a warning there that, if you make your own .wav files, they have to be 8-bit, 8kHz, would also help.  It took a while before I found that out and was able to hear my messages.

Thanks to everyone for all your work on this!

**Comment by thess** [comment]
Adding your user to the `dialout` group should be enough for RW access.

The Zoom 3095 is probably a Conexant chip - However, knowing the model and firmware version would help.
`ATI0` returns product code
`ATI3` returns firmware ID
`AT-PV` returns Conexant patch level.

Also note that the modem country code (`ATI6` or `AT+GCI?`) may not be US - modem is available in UK. See https://en.wikipedia.org/wiki/Network_Caller_ID#Modems

**Comment by usemi25** [comment]
Many thanks!  At this point, I don't think I'm going to reattach the Zoom
modem again (yes, it's a Conexant 93010, or maybe 93001, I forget, chip),
since CA is now, finally working.  Pretty sure I checked and it was the US
variant.  I was wondering about firmware version but, while I can find
instructions all over for how to upgrade, I found no firmware anywhere on
the web, so didn't even bother looking at version.

I do think it would really help people getting started to add a line to the
setup instructions about what to do to ensure CA has privileges to access
the modem, whether via chmod or adding their user ID to the dialout group.
And, while there are several places with comments about .wav files needing
to be 8 bit/8kHz in the discussion area, just 1 line noting this in the
setup instructions would have saved me a lot of time, too.

Thank you so much for your work on this.  My wife did not want to just get
rid of our Ooma and suddenly the 5 or so daily spam calls are gone and we
won't have to unplug the phone every night.

Barry
